Project2015To2017.Core/Definition/Project.cs

```diff
@@ -88,5 +88,35 @@ public DirectoryInfo NuGetPackagesPath
 		public IReadOnlyList<XElement> AssemblyAttributeProperties { get; set; } = Array.Empty<XElement>();
 
 		public IReadOnlyList<string> IntermediateOutputPaths { get; set; }
+
+		private sealed class ProjectNameFilePathEqualityComparer : IEqualityComparer<Project>
+		{
+			public bool Equals(Project x, Project y)
+			{
+				if (ReferenceEquals(x, y)) return true;
+				if (x is null) return false;
+				if (y is null) return false;
+				if (x.GetType() != y.GetType()) return false;
+				return string.Equals(x.ProjectName, y.ProjectName, StringComparison.InvariantCultureIgnoreCase) &&
+				       string.Equals(x.FilePath.FullName, y.FilePath.FullName,
+					       StringComparison.InvariantCultureIgnoreCase);
+			}
+
+			public int GetHashCode(Project obj)
```

Comment:
Why not reuse the `FilePathEqualityComparer` defined below?

Reply:
Because `FilePathEqualityComparer` is an `IEqualityComparer<Solution>`, and this takes care of `Project`. Moreover, in this `IEqualityComparer<...>` I've opted into comparing `ProjectName` too.
I could, I guess, factor out `IEqualityComparer<FileInfo>` and reuse it in both comparers, but I'm not sure it is worth the effort.

Reply:
Agreed, it was for comparing the paths yes but no big deal indeed